encrypt_util.go

```diff
@@ -298,6 +284,149 @@ func decryptStream(
 	return totalFileSize, err
 }
 
+func encryptGCM(iv []byte, plaintext []byte, encryptionKey []byte, aad []byte) ([]byte, error) {
+	aead, err := initGcm(encryptionKey)
+	if err != nil {
+		return nil, err
+	}
+	return aead.Seal(nil, iv, plaintext, aad), nil
+}
+
+func decryptGCM(iv []byte, ciphertext []byte, encryptionKey []byte, aad []byte) ([]byte, error) {
+	aead, err := initGcm(encryptionKey)
+	if err != nil {
+		return nil, err
+	}
+	return aead.Open(nil, iv, ciphertext, aad)
+}
+
+func initGcm(encryptionKey []byte) (cipher.AEAD, error) {
+	block, err := aes.NewCipher(encryptionKey)
+	if err != nil {
+		return nil, err
+	}
+	return cipher.NewGCMWithNonceSize(block, 16)
+}
+
+func encryptFileGCM(
+	sfe *snowflakeFileEncryption,
+	filename string,
+	tmpDir string) (
+	*gcmEncryptMetadata, string, error) {
+	tmpOutputFile, err := os.CreateTemp(tmpDir, baseName(filename)+"#")
```

Comment:
Why do we add "#" here?

Reply:
To be honest - I don't know, but we do the same in CBC encryption. The case is that such magic things might have solved some problems on some specific OSes or something in the past and I'm afraid to replace it.

Reply:
After trying to figure this out for some time I guess it's just a way to prevent collisions in filenames. The tempfile uses the same filename as the infile so I guess it's to prevent overriding the infile in case the tmpDir is empty or something

Reply:
Yes, I generally agree, but whether it has to be that way or not is another question. I think that in 99% we can replace it with `.bak` or anything, but I'd leave it as is.

Reply:
It will most likely not cause any problems, but in case some one, for god only know what reason, has files called my_file and my_file# it might cause a collision?

Reply:
It feels like we should at least check if a file of this name exists

Reply:
It doesn't work like that - the second argument is called `pattern`, and this is only a prefix. This function always creates a new file.